### ISSUE TITLE
Tab / auto tag completion incorrectly adds closing tag when pasting text in jsx file

### ISSUE THREAD

**Issue by joshunger** [Microsoft/vscode#62893]
Issue Type: <b>Bug</b>

1. Create a new file and save with a .jsx file extension.

2. Paste the following content:

```
<!-- Footer -->
<footer class="footer">
    <div class="container">
        <div class="has-text-centered">
            <div class="has-text-centered">
                <img class="small-footer-logo" src="assets/images/logos/bulkit-logo-g.png" alt="">
            </div>
        </div>
        <div class="has-text-centered">
            <span class="more-info-company">Startup and SaaS template</span>
        </div>
    </div>
</footer>
<!-- /Footer -->
```

3. Note the first line.

![image](https://user-images.githubusercontent.com/2301847/48301789-af3a2300-e4b0-11e8-9b5b-30e42d1c24d2.png)

Expecting: no incorrect text.

Actual: `</footer>` is incorrect added.

My use case: convert existing `.html` to a `.jsx` component. I was doing it this way -

1. Create empty `.jsx` file.
2. Paste html.
3. Wrap in `export default () => (` and `)`.
4. Clean up.

VS Code version: Code - Insiders 1.29.0-insider (5438af145cf480ce0162bda0799435684b293ec5, 2018-11-07T00:54:27.213Z)
OS version: Darwin x64 18.0.0

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i5-7267U CPU @ 3.10GHz (4 x 3100)|
|GPU Status|2d_canvas: enabled<br>checker_imaging: disabled_off<br>flash_3d: enabled<br>flash_stage3d: enabled<br>flash_stage3d_baseline: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>native_gpu_memory_buffers: enabled<br>rasterization: enabled<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|3, 3, 3|
|Memory (System)|16.00GB (0.03GB free)|
|Process Argv|.|
|Screen Reader|no|
|VM|0%|

</details><details><summary>Extensions (17)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-eslint|dba|1.7.0
xml|Dot|2.3.2
EditorConfig|Edi|0.12.5
prettier-vscode|esb|1.7.1
flow-for-vscode|flo|0.8.5
python|ms-|2018.10.0
vscode-jest|Ort|2.9.2
vscode-docker|Pet|0.3.1
java|red|0.33.0
sass-indented|rob|1.4.9
vscode-fileutils|sle|2.12.0
code-spell-checker|str|1.6.10
sort-lines|Tyr|1.7.0
vscode-java-debug|vsc|0.15.0
vscode-java-pack|vsc|0.4.0
vscode-java-test|vsc|0.10.0
vscode-maven|vsc|0.11.1

</details>

**Comment by chrmarti** [comment]
/confirm No extensions needed. Reproduced maybe not the first, but all subsequent times.

**Comment by mjbvz** [comment]
Two issue issues:

1. I fixed a bug for jsx tag complete for multi-line paste with #63178

1. That html is not valid jsx what with the comments, but the `img` tag is also not properly closed: `/>`. This causes the invalid tag completion. I've opened https://github.com/Microsoft/TypeScript/issues/28541 to track possibly improving behavior for this. Can't guarantee that will be fixed since we only officially support proper jsx 

Closing as upstream and duplicate
